java/ql/src/semmle/code/java/JDKAnnotations.qll

```diff
@@ -43,10 +39,7 @@ class TargetAnnotation extends Annotation {
    */
   Expr getATargetExpression() {
     not result instanceof ArrayInit and
-    (
-      result = this.getAValue() or
-      result = this.getAValue().(ArrayInit).getAnInit()
-    )
+    result = this.getAValue(_)
```

Comment:
Keep a no-arg `getAValue` method, since we seem to use it a lot?

Reply:
I did not remove `getAValue()`. But as mentioned in https://github.com/github/codeql/pull/6246/files#r668908394, `getAValue()` and `getAValue(string)` are actually for different use cases. The first one is for single values, the second one is for single and array values.

But while looking at it, I think the `not result instanceof ArrayInit and` in the line above can now be removed with this change.

Reply:
Actually I have marked `getAValue()` as deprecated because it is error-prone. It could by accident match the wrong annotation type element, especially when in the future new elements are added to an annotation type.